src/lib/fy-doc.c

```diff
@@ -3343,6 +3343,18 @@ enum fy_node_style fy_node_get_style(struct fy_node *fyn)
 	return fyn ? fyn->style : FYNS_PLAIN;
 }
 
+void fy_node_set_style(struct fy_node *fyn, enum fy_node_style style)
+{
+	if (!fyn)
+		return;
+
+	/* ignore alias nodes to save document structure */
+	if (fyn->style == FYNS_ALIAS)
+		return;
+
+	fyn->style = style;
```

Comment:
I wish it was so simple.

The problem is that for scalars it is quite possible to set a style that is illegal.

For instance change a double quoted scalar node's style to plain is illegal if the scalar has \0 for example.

So for the setter to work it must run a check of what are the permissible styles, and allow the setting if permitted.

Reply:
I would like to make at least unchecked dangrous version like `fy_node_set_style_unchecked`, because currently we don't have public interface to change style at all. It is a problem for outputting some nodes in somehow beautified way